### PR TITLE
Game logic

### PR DESCRIPTION
**Pull request by tta2yta**
In the third milestone section of the Tic Tac Toe game, we have created a branch called the game_logic. In this branch, we have created two classes called player and game. These classes hold instance variables and methods that represent respective objects.  In the main file, we created objects of these classes to access the needed logic implementation of the member functions in each class. This project is the implementation of the Tic Tac Toe game in the ruby language.

Kindly review my code before I merge it.
Thanks in advance.

### CODE REVIEW

**Review by bini-i — CHANGES_REQUESTED** [review]
# Status: Changes Required :yellow_circle: 
### Hello  team :wave: 

### I really like your project, 
- simple and understandable, 
-  your methods are small and do one thing only, and 
-  thank you for including a decent README early in the milestone.

#### Please work on the requirements listed below and submit a request for a new review:
- use attr_readers/attr_writers to easily access and modify attributes of your objects
eg:- name attributes of a player can be set at initialization and be accessed if you set attr_readers for it and also one player will have one name instead of an array of names.
- you can DRY repetitive code to obtain players information in main.rb by factoring them into a method.

#### If you have any question kindly reach me on slack @Binyam Hailemeskel

### HAPPY CODING :computer: :blush: 

> #### Reviewed by Binyam Hailemeskel (TSE)
> GitHub: [@bini-i](https://github.com/bini-i)
> Twitter: [@binyamshewa](https://twitter.com/binyamshewa)
> LinkedIn: [LinkedIn](https://www.linkedin.com/in/binyam-hailemeskel-728048151/)

**Comment by tta2yta** [comment]
Hello. Thank you for your feedback. I have added one attr_reader  instance variable to the class player as you said to use attr_readers/attr_writers. Moreover, to achieve the DRY principle in the main file, we have created one function called enter_names that handles the names of players.

Kindly review my code before I merge it.
Thanks in advance.

File: bin/main [hunk 1]
```diff
@@ -1,83 +1,87 @@
...
+  puts '|4| |5| |6|'
+  puts '------------'
+  puts '|7| |8| |9|'
 
-pos = [1, 2, 3, 4, 5, 6, 7, 8, 9]
+  ply = Player.new
+  #-------Player 1 Information------
+  puts 'Player 1 Enter Your Name'
+  # pl.player_name(gets.chop.strip)
+  # puts "Player One Your Name is #{pl.get_player_name(0)}"
```
Comment (lekegitrepo) [inline bin/main]: - Remove commented code.

File: bin/main [hunk 2]
```diff
@@ -1,83 +1,87 @@
...
-def check_winner?(pos)
-  return true if pos[0, 3].uniq.length == 1
-  return true if pos[3, 3].uniq.length == 1
-  return true if pos[6, 3].uniq.length == 1
-end
+  #-------Player 2 Information------
+  puts 'Player 2 Enter Your Name'
+  enter_names(1, ply)
+  # pl.player_name(gets.chop.strip)
+  # puts "Player two Your Name is #{pl.get_player_name(1)}"
```
Comment (lekegitrepo) [inline bin/main]: - Remove commented code

**Review by lekegitrepo — CHANGES_REQUESTED** [review]
## Status: Changes Required! :recycle:
### Good job :+1:  Team
#### The required changes are well implemented. However, there is commented code in the `main` file

#### commented code:
 - Remove the commented codes.

#### _Reviewed by Temitayo Adeleke_

**Comment by tta2yta** [comment]
Hello. Thank you for your code reviews. We have removed the commented codes from the main.

Kindly review my code before I merge it.
Thanks in advance.

**Review by lekegitrepo — APPROVED** [review]
## Status: Approved :heavy_check_mark:
### Good job :+1: for implementing the required changes.
#### This project is approved :white_check_mark:

#### _Reviewed by Temitayo Adeleke_

**Pull request merged**